### DISCUSSION TITLE
Authentication mechanism not supported.

### DISCUSSION THREAD

**Discussion by thewebmasterp** [umputun/remark42#1425] [Q&A] [answered]
Hi. I am trying to set up email notifications and this is my docker compose environment:
```
      - VIRTUAL_HOST=
      - LETSENCRYPT_HOST=
      - REMARK_URL=
      - SECRET=
      - SITE=
      - AUTH_EMAIL_ENABLE=true
      - AUTH_EMAIL_FROM=xxxx@xxxx.com
      - AUTH_GOOGLE_CID=
      - AUTH_GOOGLE_CSEC=
      - AUTH_GITHUB_CID=
      - AUTH_GITHUB_CSEC=
      - AUTH_FACEBOOK_CID=
      - AUTH_FACEBOOK_CSEC=
      - AUTH_TWITTER_CID=
      - AUTH_TWITTER_CSEC=
      - NOTIFY_USERS=email
      - NOTIFY_ADMINS=email
      - NOTIFY_EMAIL_FROM=xxxx@xxxx.com
      - ADMIN_SHARED_EMAIL=yyyy@xxxx.com
      - SMTP_HOST=smtp.dynu.com
      - SMTP_PORT=465
      - SMTP_TLS=true
      - SMTP_USERNAME=xxxx@xxxx.com
      - SMTP_PASSWORD=xxxxxxxxxx
```
I have email hosting at dynu.com. Created an email account there and trying to connect it to remark42 via SMTP, to make it send notifications from this email account but it doesn't work. Here's the error I get:

```
remark42             | 2022/07/21 06:03:56.989 [WARN]  failed to send to email: with username 'notify@xxxx.com' at server smtp.dynu.com:465 with TLS, 1 error occurred:
remark42             | 	* problem sending admin email notification to "y@xxxx.com": failed to auth to smtp smtp.dynu.com:465, 504 Authentication mechanism not supported.

```

I tried all the 25, 26, 2525, 465, 587 ports but it doesn't work. Tried toggling the SMTP_TLS to no avail (getting some different errors, seemingly worse than this one). From all I read online, It should be port 465 and SMTP_TLS needs to be set to true.

I know it's a remark42 issue because I connect an email account hosted at dynu without any problem to my email client K-9 mail and it works perfectly fine.

I compared both configurations, K-9's and remark's and the difference I see is in the protocol. While K-9 allows for both TLS/SSL (and it works), in remark42 due to its variable SMTP_TLS=true I see that it probably tries only TLS, rather than TLS/SSL. This seems to be the issue in my opinion.

Thanks!

**Comment by paskal** [comment]
Please try port 587 with SMTP_TLS=false and SMTP_STARTTLS=true.

**Reply by thewebmasterp** [reply]
Thanks for the suggestion but I get the same error :(
```
remark42             | 	* problem sending admin email notification to "my@email.com": failed to auth to smtp smtp.dynu.com:587, 504 Authentication mechanism not supported.

```

**Reply by paskal** [reply]
It's possible to troubleshoot what's wrong here precisely, but it's unlikely we will alter the remark42 to support a single faulty provider, as it seems to me that most of the providers already work without problems, namely MailChimp, SendGrid, and multiple local email relays like ssmtp.

I would propose contacting the support of the service you are paying for so they would troubleshoot the issue on their side and make sure they support the auth standards supported by the rest of the companies on the market.

**Comment by thewebmasterp** [comment, marked as answer]
I resolved the issue by putting Google in the middle, despite my hate for big tech corps, I had no other choice.

Created a google account. Added my dynu email account as a secondary email to google and made it the default sending mail (Sending mail from this gmail account will by default use my dynu custom email). Then simply connected the gmail to remark42.

Through the process there was only a single issue. POP3 access to Gmail account is no longer possible, to bypass this simply set up 2fa on your google account and use an app password.
